packages/@react-spectrum/menu/src/SubMenuTrigger.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, useIsMobileDevice} from '@react-spectrum/utils';
+import {MenuContext, SubMenuTriggerContext, useMenuStateContext} from './context';
+import {Popover} from '@react-spectrum/overlays';
+import React, {Key, ReactElement, useRef} from 'react';
+import ReactDOM from 'react-dom';
+import styles from '@adobe/spectrum-css-temp/components/menu/vars.css';
+import {UNSTABLE_useSubMenuTrigger} from '@react-aria/menu';
+import {UNSTABLE_useSubMenuTriggerState} from '@react-stately/menu';
+
+// TODO: Add shouldFlip and closeOnSelect if we feel like those should be customizable for SubMenuTrigger
+// Other MenuTriggerProps like onOpenChange and positioning stuff have been removed as per discussion
+interface SubMenuTriggerProps {
+  /**
+   * The contents of the SubMenuTrigger - a Item and a Menu.
+   */
+  children: ReactElement[],
+  targetKey: Key
+}
```

Comment:
Open to opinions, would we want `closeOnSelect` to be specific to each submenu? Not sure of a good use case there